content/docs/user-guide/gems/reference/multiplayer/multiplayer/tutorial_multiplayer.md

```diff
@@ -0,0 +1,887 @@
+    {
+        AZ_Printf( "MyFirstNetworkComponent", "client = %f", uptime );
+    }
+}
+```
+
+### Network Component Controllers
+
+Let's improve on the separation between server and client code. At the moment, `MyFirstNetworkComponent` performs both server and client duties. Let's break it up. In O3DE Multiplayer, the way to do that is by using `controllers`. A controller does not exist on `Client` roles but does exist on Authority role. We already saw a glipmse into that in our previous examples:
+
+```c++
+auto* controller = static_cast<MyFirstNetworkComponentController*>( GetController() );
+```
+
+Going back to XML, the controller were mentioned in the attribute `OverrideController`:
+
+```xml
+<Component Name="MyFirstNetworkComponent"
+           Namespace="MultiplayerSample"
+           OverrideComponent="true"
+           OverrideController="false"
+```
+
+Now that we wish to do some custom work with the controller, we will flip `OverrideController` to true.
+
+```xml
+<Component Name="MyFirstNetworkComponent"
+           Namespace="MultiplayerSample"
+           OverrideComponent="true"
+           OverrideController="true"
+```
+
+When attempt to build the code, you will get compile errors but that is because the code now expects us to override the controller base class. In fact, if you look into the code generated header, `D:\git\o3de\build\-86ba765d\Gem\Code\Azcg\Generated\Source\AutoGen\MyFirstNetworkComponent.AutoComponent.h`, like we did before, you will see that large comment has changed to add controller:
+
+```c++
+/// You may use the classes below as a basis for your new derived classes. Derived classes must be marked in MyFirstNetworkComponent.AutoComponent.xml
+/*
+    class MyFirstNetworkComponentController
+        : public MyFirstNetworkComponentControllerBase
+    {
+    public:
+        MyFirstNetworkComponentController(MyFirstNetworkComponent& parent);
+
+        void OnActivate(Multiplayer::EntityIsMigrating entityIsMigrating) override;
+        void OnDeactivate(Multiplayer::EntityIsMigrating entityIsMigrating) override;
+    };
+
+    MyFirstNetworkComponentController::MyFirstNetworkComponentController(MyFirstNetworkComponent& parent)
+        : MyFirstNetworkComponentControllerBase(parent)
+    {
+    }
+
+    void MyFirstNetworkComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void MyFirstNetworkComponentController::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+*/
+```
+
+Let's add this content to our existing `D:\git\o3de-multiplayersample\Gem\Code\Source\Components\MyFirstNetworkComponent.h` and `D:\git\o3de-multiplayersample\Gem\Code\Source\Components\MyFirstNetworkComponent.cpp`.
+
+Once you do, you can move then move all the server logic into `MyFirstNetworkComponentController` and keep client logic in `MyFirstNetworkComponent`. Here is the result.
```

Comment:
One you do, you can move all the server logic...

Reply:
Fixing.